tests/MongoDB.Driver.Core.Tests/Core/Authentication/ScramSha256AuthenticatorTests.cs

```diff
@@ -261,5 +261,62 @@ public void Constructor_should_throw_an_ArgumentNullException_when_credential_is
                            conversationId: 1, " + 
                 $"         payload: new BinData(0, \"{ToUtf8Base64(_clientRequest2)}\")}}}}");
         }
+
+        [Theory]
+        [ParameterAttributeData]
+        public void Authenticate_should_use_cache(
+            [Values(false, true)] bool async)
+        {
+            var randomStringGenerator = new ConstantRandomStringGenerator(_clientNonce);
+            var subject = new ScramSha256Authenticator(__credential, randomStringGenerator);
+
+            var saslStartReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                @"{conversationId: 1," +
+                $" payload: BinData(0,\"{ToUtf8Base64(_serverResponse1)}\")," +
+                @" done: false,
+                   ok: 1}"));
+            var saslContinueReply = MessageHelper.BuildReply<RawBsonDocument>(RawBsonDocumentHelper.FromJson(
+                @"{conversationId: 1," +
+                $" payload: BinData(0,\"{ToUtf8Base64(_serverReponse2)}\")," +
+                @" done: true,
+                   ok: 1}"));
+
+            var connection = new MockConnection(__serverId);
+            connection.EnqueueReplyMessage(saslStartReply);
+            connection.EnqueueReplyMessage(saslContinueReply);
+
+            if (async)
+            {
+                subject.AuthenticateAsync(connection, __description, CancellationToken.None).GetAwaiter()
+                    .GetResult();
+            }
+            else
+            {
+                subject.Authenticate(connection, __description, CancellationToken.None);
+            }
+
+            SpinWait.SpinUntil(() => connection.GetSentMessages().Count >= 2, TimeSpan.FromSeconds(5)).Should()
+                .BeTrue();
+
+            subject._cache().Should().NotBe(null);
+            subject._cache()._cacheKey().Should().NotBe(null);
+            subject._cache()._cacheEntry().Should().NotBe(null);
+        }
+    }
+
+    internal static class ScramShaAuthenticatorReflector
+    {
+        public static ScramCache _cache(this ScramShaAuthenticator obj) =>
+             (ScramCache)Reflector.GetFieldValue(Reflector.GetFieldValue(obj, "_mechanism"), nameof(_cache));
+
+    }
+
+    internal static class ScramCacheReflector
+    {
+        public static ScramCacheKey _cacheKey (this ScramCache obj) =>
+            (ScramCacheKey)Reflector.GetFieldValue(obj, nameof(_cacheKey));
+
+        public static ScramCacheEntry _cacheEntry (this ScramCache obj) =>
+            (ScramCacheEntry)Reflector.GetFieldValue(obj, nameof(_cacheEntry));
```

Comment:
Didn't you rename this to `_cachedEntry`? (with a "d")

Reply:
Yes, just fixed that.